router/core/header_rule_engine.go

```diff
@@ -136,6 +138,31 @@ func NewHeaderPropagation(rules *config.HeaderRules) (*HeaderPropagation, error)
 	return &hf, nil
 }
 
+func (hf *HeaderPropagation) convertCacheToRules(cacheControl config.CacheControlPolicy) {
```

Comment:
This is the direction I found simplest - to simply convert the cache control policy object into the appropriate rules. It isn't the only way, and I'm happy to get push-back on it, I had just thought it was the simplest and fastest way of adding in the functionality